### ISSUE TITLE
normalization in cosine similarity

### ISSUE THREAD

**Issue by ngc92** [tensorflow/tensorflow#33820]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/losses/cosine_similarity
https://www.tensorflow.org/api_docs/python/tf/keras/losses/CosineSimilarity

## Description of issue (what needs changing):
The documentation for the cosine similarity does not state whether `y_true` and `y_pred`
are expected to be normalized vectors. The provided equation `loss = -sum(y_true * y_pred)`
suggests the need to be, but looking at the source, they are normalized as part of the computation.
```
y_true = nn.l2_normalize(y_true, axis=axis)
y_pred = nn.l2_normalize(y_pred, axis=axis)
return -math_ops.reduce_sum(y_true * y_pred, axis=axis)
```
As a special case, the doc does not state what happens in the case of either being zero.

(Also, isn't the above implementation suboptimal in terms of speed, as each element is divided by 
the norm, instead of simply dividing the result once?)

**Comment by jvishnuvardhan** [comment]
@ngc92 Can you contribute through a PR to update the docs? Thanks!

**Comment by MarkDaoust** [comment]
Yes, can you send a PR?

The code is here:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/losses.py#L1097-L1141

If you make a PR please use [doctest format](https://docs.python.org/3/library/doctest.html), for example code so that it is testable (prefix code lines with `>>>`).

**Comment by Ir1d** [comment]
I think this could be closed since #37090 is merged ?

**Comment by jvishnuvardhan** [comment]
@Ir1d Thanks for the note. I am closing this issue as the PR https://github.com/tensorflow/tensorflow/pull/37090 related to this issue already merged. Thanks!

**Comment by ngc92** [comment]
The documentation is still lacking for the `CosineSimilarity` class, and contains a sign error. 
I have submittd #37343 for that.
